### PR TITLE
feat: Upgrade Annotation Engine

### PR DESCRIPTION
**Pull request by Kerwin-L**
Add ToolScheduler to switch different tool

### CODE REVIEW

File: packages/lb-annotation/src/core/toolOperation/basicToolOperation.ts [hunk 1]
```diff
@@ -675,6 +675,13 @@ class BasicToolOperation extends EventListener {
     this.forbidCursorLine = false;
   }
 
+  public clearCursorLine() {
```
Comment (Glenfiddish) [inline packages/lb-annotation/src/core/toolOperation/basicToolOperation.ts:678]: What's diff between forbidCursorLine and clearCursorLine ?

File: packages/lb-annotation/src/core/scheduler.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  }
+
+  public getEmptyImage(width: number, height: number) {
+    const imgSrc = createEmptyImage({ width, height });
+    const emptyImgNode = new Image();
+    emptyImgNode.src = imgSrc;
+    return emptyImgNode;
+  }
+
+  public createOperation(tool?: EToolName, imgNode?: HTMLImageElement, config?: Object) {
```
Comment (Glenfiddish) [inline packages/lb-annotation/src/core/scheduler.ts:144]: Add comment for this function

File: packages/lb-annotation/src/core/toolOperation/basicToolOperation.ts (hunk 1 above)
Comment (Kerwin-L) [reply packages/lb-annotation/src/core/toolOperation/basicToolOperation.ts:678]: `ForbidCursorLine` is always not showing  cursorLine. But `ClearCursorLine` just move coordinate out of view to hidden cursorLine

**Pull request merged**